### PR TITLE
Cognoma feature selection using PCA (2000 features)

### PR DESCRIPTION
**Pull request by mans2singh**
Hey Folks:

First cut of feature selection using PCA (2000 features).   Let me know your comments/recommendations.

Thanks

### CODE REVIEW

File: algorithms/scripts/SGDClassifier-PCA-mans2singh.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+
+# ## PCA for feature selection
+
+# In[40]:
+
+# select the top features with the PCA
+feature_select = PCA(n_components=n_feature_kept)
```
Comment (dhimmel) [inline algorithms/scripts/SGDClassifier-PCA-mans2singh.py]: Should this be renamed to `feature_transform` because we're doing more than just selection?

**Review by dhimmel — CHANGES_REQUESTED** [review]
Looks good to me besides my single comment

**Comment by mans2singh** [comment]
@dhimmel - Thanks for your comments.  I've updated the variable to feature_transform.  Let me know if you have any other recommendation. 

Thanks

**Comment by dhimmel** [comment]
@mans2singh, nice work. Performance looks good: testing AUROC = 93.1% and training AUROC = 95.2%. I don't think we can directly compare it to not using PCA, since I didn't see another notebook that used `n_feature_kept = 2000`.

I'm going to merge now. However, if you'd like to keep working on this, you can submit another pull request with additional modifications. Future directions could be
1. comparing PCA to MAD feature selection or no feature selection
2. plotting cumulative explained variance versus number of components

**Review by dhimmel — APPROVED** [review]

**Comment by mans2singh** [comment]
Thanks @dhimmel

**Pull request merged**